### PR TITLE
Update: gtk_doc

### PR DESCRIPTION
**Pull request by ThatGeekyWeeb**
I hate to open a WIP PR, but I am having a *very* hard time getting things to work, so, I'd love some help!
***
What's being done:
`gtk_doc` is going from `1.29` to `1.32`
Which requires `DocBook XML DTD V4.3`
> Our version of `docbook_xml` is too new

So a new package for `docbook_xml`(`43`) version 4.3 needs to be made, I did that, but, compiling for `gtk_doc` still fails-
~
Additionally: 
`xmlcatmgr` has been added and will ease the editing of `catalog.xml`\
And there are **alot** of patches that need to be applied to `gtk_doc`, and the way I have it set up to grab and apply these patches isn't ideal.
***
For reference, here is `xpbs-src`'s template for `docbook-xml` which installs `4.5`, `4.4`, `4.3`, and `4.2`
> https://github.com/void-linux/void-packages/blob/master/srcpkgs/docbook-xml/template

### CODE REVIEW

**Comment by uberhacker** [comment]
No problem.  Just remove the `[WIP]` when you are ready.

**Comment by vincowl** [comment]
@ThatGeekyWeeb I've just tried to compile gtk_doc you provided, but ended with an error with docbook_xml43 : 
```
docbook_xml43 failed to install: `xmlcatmgr -c /usr/local/etc/xml/catalog.xml add rewriteSystem http://www.oasis-open.org/docbook/xml/4.3 /usr/local/share/xml/docbook/xml-dtd-4.3` exited with 127
```

**Comment by ThatGeekyWeeb** [comment]
@vincowl, irrc exit code `127` means that the command doesn't exist, which means `xmlcatmgr` is missing. Install `xmlcatmgr` from this PR. If you already have, then I'll have to do some debugging.

**Comment by vincowl** [comment]
Indeed, I was trusting the dependency toolchain.... Bad idea ! OK then. I successfully installed docbook_xml43 but configure step of gtk_doc fails with : 
```
checking for XML catalog... /usr/local/etc/xml/catalog.xml
checking for xmlcatalog... /usr/local/bin/xmlcatalog
checking for DocBook XML DTD V4.3 in XML catalog... not found
configure: error: could not find DocBook XML DTD V4.3 in XML catalog
```
Are you facing the same error ?

**Comment by ThatGeekyWeeb** [comment]
@vincowl, yes that is the same issue I am having.

**Comment by ThatGeekyWeeb** [comment]
I figured out that `./configure` runs `xmlcatalog /usr/local/etc/xml/catalog.xml -//OASIS//DTD DocBook XML V4.3//EN` to check for `DocBook XML DTD V4.3`, if the command does not exit with code 0, then it fails, I'll have to do some debugging

**Comment by ThatGeekyWeeb** [comment]
I found that `catalog.xml` contains 
```xml
<delegatePublic publicIdStartString="-//OASIS//ENTITIES DocBook XML" catalog="file:///usr/local/etc/xml/docbook"/>
```
Which means `xmlcatalog` redirects `-//OASIS//DTD DocBook XML*` searches to `/usr/local/etc/xml/docbook`, and then checks for a result.
I added 
```xml
  <public publicId="-//OASIS//DTD DocBook XML V4.3//EN" uri="file:///usr/local/share/xml/dockbook/4.3/catalog.xml"/>
```
into `/usr/local/etc/xml/docbook` and `xmlcatalog /usr/local/etc/xml/catalog.xml "-//OASIS//DTD DocBook XML V4.3//EN"` run successfully! I'll push a fix soon!

**Comment by ThatGeekyWeeb** [comment]
Please reinstall `docbook_xsl` before reinstalling `docbook_xml43`

**Comment by ThatGeekyWeeb** [comment]
Having another package for `docbook-xml` v4.3 is bad, and `docbook_xml` should be edited to install 5.0, 5.1, 5.2, 4.5, 4.4, 4.3, and 4.2.
*** 
Additionally `docbook_xml` needs to be updated to 5.2

**Comment by ThatGeekyWeeb** [comment]
Additionally `gtk_doc` compiles fine now!

**Comment by vincowl** [comment]
Great !!

While php concurrent versions is bad, why would it be a problem to have several xml DTDs available ? Just be sure to provide access to both 4.3 and 5.0. It's always a pain to have to stick to an old version due to strong dependency to a slow development package.

If those are mutually exclusive, then have a look at php73, php74 and the upper layer php.

File: packages/docbook_xml43.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  source_url 'http://www.oasis-open.org/docbook/xml/4.3/docbook-xml-4.3.zip'
+  source_sha256 '23068a94ea6fd484b004c5a73ec36a66aa47ea8f0d6b62cc1695931f5c143464'
+
+  depends_on 'docbook'
+  depends_on 'xmlcatmgr'
+  depends_on 'docbook_xml'
+  depends_on 'docbook_xsl' # Requires the catalog.xml created within this package
+  
+  def self.prebuild
+    system "sed -i -e 's,<!-- .* -->,,g' /usr/local/etc/xml/catalog.xml"
```
Comment (vincowl) [inline packages/docbook_xml43.rb:17]: Use `#{CREW_PREFIX}` instead of `/usr/local`

File: packages/docbook_xml43.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    system 'xmlcatmgr -c /usr/local/etc/xml/docbook add public "-//OASIS//DTD DocBook XML V4.3//EN" "/usr/local/share/xml/docbook/4.3/catalog.xml"'
```
Comment (vincowl) [inline packages/docbook_xml43.rb:18]: Use `#{CREW_PREFIX}` instead of `/usr/local`

**Comment by ThatGeekyWeeb** [comment]
I was planning on turning `docbook_xml` into a `is_fake` package that installs 4.2, 4.2, 4.3, 4.5, 5.0, and 5.1, since that's what most distros seem to do.

File: packages/gtk_doc.rb [hunk 3]
```diff
@@ -1,38 +1,51 @@
...
+    puts
+  end
   def self.build
-    system './autogen.sh',
-           "--prefix=#{CREW_PREFIX}",
-           "--libdir=#{CREW_LIB_PREFIX}",
-           "--with-xml-catalog=#{CREW_PREFIX}/etc/xml/catalog"
-    system 'make'
+      system "./configure #{CREW_OPTIONS} --with-xml-catalog=#{CREW_PREFIX}/etc/xml/catalog.xml"
+      system "make -j#{CREW_NPROC}"
```
Comment (vincowl) [inline packages/gtk_doc.rb:46]: `-j#{CREW_NPROC}` is useless as this value is already defined by default in package.rb. Can become `system 'make'`

File: packages/xmlcatmgr.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  homepage 'https://xmlcatmgr.sourceforge.net'
+  version '2.2'
+  compatibility 'all'
+  source_url 'https://downloads.sourceforge.net/sourceforge/xmlcatmgr/xmlcatmgr-2.2.tar.gz'
+  source_sha256 'ea1142b6aef40fbd624fc3e2130cf10cf081b5fa88e5229c92b8f515779d6fdc'
+
+
+  def self.build
+      system "./configure #{CREW_OPTIONS} "
+      system "make -j#{CREW_NPROC}"
```
Comment (vincowl) [inline packages/xmlcatmgr.rb:14]: `-j#{CREW_NPROC}` is useless as this value is already defined by default in package.rb

**Comment by vincowl** [comment]
Sounds great to me 👍

File: packages/docbook_xml43.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  source_sha256 '23068a94ea6fd484b004c5a73ec36a66aa47ea8f0d6b62cc1695931f5c143464'
+
+  depends_on 'docbook'
+  depends_on 'xmlcatmgr'
+  depends_on 'docbook_xml'
+  depends_on 'docbook_xsl' # Requires the catalog.xml created within this package
+  
+  def self.prebuild
+    system "sed -i -e 's,<!-- .* -->,,g' #{CREW_PREFIX}/etc/xml/catalog.xml"
+    system "xmlcatmgr -c #{CREW_PREFIX}/etc/xml/docbook add public '-//OASIS//DTD DocBook XML V4.3//EN' '#{CREW_PREFIX}/share/xml/docbook/4.3/catalog.xml'"
```
Comment (uberhacker) [inline packages/docbook_xml43.rb:18]: Does this command generate an output file?  If so, should this be an install step?

Comment (ThatGeekyWeeb) [reply packages/docbook_xml43.rb:18]: It edits a file installed by `docbook_xml`
See #4315

File: packages/gtk_doc.rb [hunk 6]
```diff
@@ -1,38 +1,51 @@
...
+  depends_on 'docbook_xml'
+  depends_on 'docbook_xsl'
+  depends_on 'itstool'
   depends_on 'libxslt'
+  depends_on 'docbook_xml43'
 
+  def self.patch
+    puts
+    puts 'Grabbing patches'.lightblue
+    system 'curl --ssl --insecure -L -o "output-reproducible.patch" "https://git.io/JUlWD" --progress-bar'
```
Comment (uberhacker) [inline packages/gtk_doc.rb:22]: Why do we need the `--insecure` option for https?

Comment (ThatGeekyWeeb) [reply packages/gtk_doc.rb:22]: We don't idk exactly why I have that there, I'll remove that.

File: packages/docbook_xml43.rb (hunk 5 above)
Comment (uberhacker) [reply packages/docbook_xml43.rb:18]: Should this command be executed in `def self.preinstall` instead?

Comment (ThatGeekyWeeb) [reply packages/docbook_xml43.rb:18]: Yes, that would be a better place, as nothing is actually built.

**Comment by ThatGeekyWeeb** [comment]
reinstallation of `docbook_xml43` has a 100% chance to fail, as a file provided by another package it edited, I've added a bash script, that is both made and used inside `docbook_xml43.rb`, that will allow it to reinstall.
***
> Another idea would be to add a way for commands to fail, only if allowed.
> 
> ```
> xmlcatmgr -c /usr/local/etc/xml/docbook remove public "-//OASIS//DTD DocBook XML V4.3//EN"
> ```
> 
> Fails unless `docbook_xml43` has been installed once before, but that's fine, as long as
> 
> ```
> xmlcatmgr -c #{CREW_PREFIX}/etc/xml/docbook add public '-//OASIS//DTD DocBook XML V4.3//EN' '#{CREW_PREFIX}/share/xml/docbook/4.3/catalog.xml'"
> ```
> 
> runs afterwards its fine. But due to how `crew` works, it end as soon as one command fails.
> ***
> A final option is to use a bash script, which would execute both of them, and exit with `0` no matter what.

Please see #4315 for more info.\
~\
`/usr/local` must be used, as it is writing to a bash script.

File: packages/docbook_xml43.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  version '4.3'
+  source_url 'http://www.oasis-open.org/docbook/xml/4.3/docbook-xml-4.3.zip'
+  source_sha256 '23068a94ea6fd484b004c5a73ec36a66aa47ea8f0d6b62cc1695931f5c143464'
+
+  depends_on 'docbook'
+  depends_on 'xmlcatmgr'
+  depends_on 'docbook_xml'
+  depends_on 'docbook_xsl' # Requires the catalog.xml created within this package
+  
+  def self.preinstall
```
Comment (uberhacker) [inline packages/docbook_xml43.rb:16]: Should this be `def self.prebuild`?

File: packages/docbook_xml43.rb [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    system "sed -i -e 's,<!-- .* -->,,g' #{CREW_PREFIX}/etc/xml/catalog.xml"
+    system "cat << EOF > ./remove_add.sh
+xmlcatmgr -c /usr/local/xml/docbook remove public '-//OASIS//DTD DocBook XML V4.3//EN' '/usr/local/share/xml/docbook/4.3/catalog.xml'
+xmlcatmgr -c /usr/local/etc/xml/docbook add public '-//OASIS//DTD DocBook XML V4.3//EN' '/usr/local/share/xml/docbook/4.3/catalog.xml'
```
Comment (uberhacker) [inline packages/docbook_xml43.rb:20]: Please abstract `/usr/local` with `#{CREW_PREFIX}`.

**Comment by uberhacker** [comment]
Nice work!

**Pull request merged**